### DISCUSSION TITLE
-1 ids when searching - what is the right way to fix this and not get inf scores?

### DISCUSSION THREAD

**Discussion by drei34** [facebookresearch/faiss#3721] [Q&A]
Hi I am using Vespa and faiss python to do some experiments and Vespa seems to pick up the correct ann while the python library does not, it is quite off so the HNSW inside of this library is giving me very different results.

Specifically, I have 1.2M vectors and 14 or so query vectors. For the first query vector, I have some document vectors that are an L2 distance 0.44 away but FAISS returns the ann as one which has distance 0.58 away (the best one). When I also make my k in kNN be 1.2 M I get many -1 in the index and many +inf distances in the distance matrix.

What parameters can I modify to get more results? I understand it will be slower. Is there something to do?

**Comment by mdouze** [comment]
The speed-accuracy tradeoff parameter for HNSW is `efSearch`. 
https://github.com/facebookresearch/faiss/wiki/Guidelines-to-choose-an-index#if-not-hnswm-or-ivf1024pqnx4fsrflat
